src/middlewared/middlewared/plugins/failover_/remote.py

```diff
@@ -192,8 +192,12 @@ def get_remote_os_version(self):
         if self.client is not None and self._remote_os_version is None:
             try:
                 self._remote_os_version = self.client.call('system.version')
-            except Exception:
-                logger.error('Failed to determine OS version', exc_info=True)
+            except Exception as e:
+                if "'NoneType' object has no attribute 'text_message'" in str(e):
```

Comment:
Won't it be better to handle this in the middleware client itself and reraise a specific exception?

Reply:
Just getting back to this and the RemoteClient was already doing the hard-work of catching these specific exceptions so I just need to do the same in `get_remote_os_version`. I've made the correct changes in latest PR.